lib/bolt/transport/api.rb

```diff
@@ -11,6 +11,12 @@ def initialize(*args)
         super
       end
 
+      def conn_plan_job(plan_id)
```

Comment:
Another question: Do these connections get reused for multiple plan runs? Should the plan id be passed in to each `build_request` and `finish_plan` call so it doesn't get overwritten?

Reply:
yes the executor is shared when puma is threaded. You'll need to create a separate executor for each request.